src/Navigation.re

```diff
@@ -14,6 +14,25 @@ external navigateWithParams: (t, string, Js.t({..})) => unit = "navigate";
 [@bs.send] external goBackToRoute: (t, string) => unit = "";
 [@bs.send] external getParam: (t, string) => Js.nullable('a) = "";
 [@bs.send] external getParamWithDefault: (t, string, 'a) => 'a = "getParam";
+[@bs.send]
+external addListener:
+  (
+    t,
+    string,
+    {
+      .
+      "action": {
+        .
+        "type": string,
+        "key": string,
+      },
+      "context": string,
+      "type": [ | `didBlur | `didFocus | `willBlur | `willFocus],
+    } =>
+    unit
+  ) =>
+  unit =
```

Comment:
The return type is incorrect. The `addListener` function returns a JS object containing a `remove` function so that the subscription can be cancelled again.